plugins/nf-amazon/src/main/com/upplication/s3fs/S3OutputStream.java

```diff
@@ -641,14 +646,14 @@ int getPartsCount() {
      */
     static synchronized ExecutorService getOrCreateExecutor(int maxThreads) {
         if( executorSingleton == null ) {
-            ThreadPoolExecutor pool = new ThreadPoolExecutor(
-                    maxThreads,
-                    Integer.MAX_VALUE,
-                    60L, TimeUnit.SECONDS,
-                    new LimitedQueue<Runnable>(maxThreads *3),
-                    new ThreadPoolExecutor.CallerRunsPolicy() );
-
-            pool.allowCoreThreadTimeOut(true);
+            ThreadPoolExecutor pool = new ThreadPoolBuilder()
+                    .withName("S3OutputStream")
+                    .withMinSize(maxThreads)
+                    .withMaxSize(maxThreads)
```

Comment:
Would not be better to use [this](https://github.com/nextflow-io/nextflow/blob/a06b444236164b88218ef548f7bf834727faa33f/modules/nextflow/src/main/groovy/nextflow/util/ThreadPoolBuilder.groovy#L163-L177)

Reply:
didn't realize there are these static methods, I'll use them

only concern is about `withKeepAliveTime` and `withAllowCoreThreadTimeout` feature. May I need to include them as arguments in the static methods maybe with some default values?

Reply:
Is this still a concern? don't think impact much these settings

Reply:
Yes, I agree, it seems doesn't impact so much, at least with my "normal" connection all tests were fine. In some test I could see 2 or 3 retries due timeout but executor retried them and all go fine